lib/sycamore/sycamore/data/table.py

```diff
@@ -0,0 +1,164 @@
+from collections import OrderedDict
+from dataclasses import dataclass, field
+from typing import Any, Optional
+import xml.etree.ElementTree as ET
+
+from sycamore.data import BoundingBox
+from PIL import Image, ImageDraw
+import numpy as np
+from pandas import DataFrame
+
+
+# This data model is similar to that used by Textract and TableTransformers.
+# It is intended to be a common intermediate representation for a variety of
+# table structure recognition models.
+@dataclass(frozen=True)
+class TableCell:
+    content: str
+    rows: list[int]
+    cols: list[int]
+    is_header: bool = False
+    bbox: Optional[BoundingBox] = None
+    # Model/format specific properties
+    properties: dict[str, Any] = field(default_factory=dict)
+
+
+class Table:
+    def __init__(self, cells: list[TableCell], caption: Optional[str] = None):
+        self.cells: list[TableCell] = cells
+        self.caption = caption
+        self.num_rows = max(max(c.rows) for c in self.cells) + 1
+        self.num_cols = max(max(c.cols) for c in self.cells) + 1
+
+    def __eq__(self, other):
+        if type(other) is not type(self):
+            return False
+
+        if self.num_rows != other.num_rows or self.num_cols != other.num_cols:
+            return False
+
+        # TODO: Yikes! this is expensive. Do the sorting in init or something.
```

Comment:
Yes!  Canonical representation seems like a good invariant.

Reply:
I can move sorting to init. I'm debating whether the table should be immutable, in which case that's fine. The only caveat is that the models do a bunch of manipulation, so it may impact how early I convert to this Table object vs how many copies are required/etc.